### PR TITLE
Board refactoring

### PR DESCRIPTION
**Pull request by cnak**

### CODE REVIEW

File: src/tic_tac_toe/board.clj [hunk 1]
```diff
@@ -16,39 +16,35 @@
...
-                        [2 5 8]
-                        [0 3 6]
-                        [0 4 8]
-                        [2 4 6]])
+(defn- gen-diagonals-left [board]
+  (let [step-by (Math/round (Math/sqrt (count board)))]
+  (vec (range (- step-by 1) (- (count board) 1) (- step-by 1)))))
+
+(defn- gen-diagonals-right [board]
+  (let [step-by (+ 1 (Math/round (Math/sqrt (count board))))]
```
Comment (c-j-j) [inline src/tic_tac_toe/board.clj]: why is Math/round required here?

File: src/tic_tac_toe/board.clj [hunk 2]
```diff
@@ -16,39 +16,35 @@
...
-
-(def winning-positions [[0 1 2] 
-                        [3 4 5] 
-                        [6 7 8]
-                        [1 4 7]
-                        [2 5 8]
-                        [0 3 6]
-                        [0 4 8]
-                        [2 4 6]])
+(defn- gen-diagonals-left [board]
```
Comment (c-j-j) [inline src/tic_tac_toe/board.clj]: not sure on what the `gen` prefix adds here. I would not abbreviate words when doing naming as it makes it a little ambigious as to what "gen" means.

If they are just the diagonal positions, I would just cal them `diagonals-left`, `columns`, etc.

File: src/tic_tac_toe/board.clj [hunk 3]
```diff
@@ -16,39 +16,35 @@
...
+  (let [row-length (Math/round (Math/sqrt (count board)))]
+    (vec (map #(vec  (range % (+ % row-length) 1)) (range 0 (count board) row-length)))))
+
+(defn- win-positions [board]
+  (into (gen-diagonals board) (into (gen-rows board) (gen-columns board))))
 
 (defn- all-cell-same? [board line] (cells-the-same? #(get-cells % line) board)) 
 
+(defn- any? [coll] 
+  (if (some true? coll) true false))
```
Comment (c-j-j) [inline src/tic_tac_toe/board.clj:48]: don't need an `if` here.. just return the output of `(some true? coll)`

File: src/tic_tac_toe/board.clj [hunk 4]
```diff
@@ -72,9 +68,7 @@
   (let [predicate #(= empty-mark %) newboard board] (keep-indexed (fn [i x] (when (predicate x) i))newboard)))
 
 (defn game-won? [board]
-  (if (= 16 (count board)) 
-  (winner? board winning-positions-four-by-four)
-  (winner? board winning-positions)))
+    (winner? board (win-positions board)))
```
Comment (c-j-j) [inline src/tic_tac_toe/board.clj]: is there a need for both `game-won?` and `winner?` Can this not just be a single method that is used? It'll be confusing for anyone using the API to decide which one to use.

File: src/tic_tac_toe/board.clj (hunk 3 above)
Comment (cnak) [inline src/tic_tac_toe/board.clj:48]: In the case of a false condition, 'some' will return a 'nill' value back.

**Pull request merged**